spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -372,8 +372,48 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
     this
   }
 
-  def cloneModule(): AbstractModule[A, B, T] = {
-    SerializationUtils.clone(this)
+  def cloneModule(deepCopy : Boolean = true): AbstractModule[A, B, T] = {
+    val moduleData = ModuleData[T](this.
+      asInstanceOf[AbstractModule[Activity, Activity, T]], Seq[String](), Seq[String]())
+    val serializedModule = ModuleSerializer.serialize[T](moduleData, false)
+    val copy = ModuleSerializer.load[T](serializedModule, false).module
+      .asInstanceOf[AbstractModule[A, B, T]]
+    setWeightAndBias(copy, deepCopy)
+    copy
+  }
+
+  private def setWeightAndBias(copy : AbstractModule[A, B, T], deepCopy : Boolean): Unit = {
+    val parameterTable = this.getParametersTable
+    val copiedModuleParamTable = copy.getParametersTable
+    if (parameterTable != null) {
+      require(copiedModuleParamTable != null, "cloned module should have params")
+      parameterTable.foreach {
+        case (name: String, params: Table) =>
```

Comment:
@i8run This assumes that each element in `getParametersTable` is a Table. Please make sure it works with the quantized module

Reply:
I discussed offline with @i8run ,  `getParamtersTable` works for quantized module, the only difference is the weight could be array thus I added the support for array paramters

Reply:
The array is because of the groups weight in convolution. It can't be converted to one tensor, because it's relevant with implementation of native code and the performance.